pkg/manager/manager_test.go

```diff
@@ -296,30 +297,48 @@ var _ = Describe("manger.Manager", func() {
+				m, err := New(cfg, Options{
+					newResourceLock: func(_ *rest.Config, _ recorder.Provider, _ leaderelection.Options) (resourcelock.Interface, error) {
+						return nil, fmt.Errorf("expected error")
+					},
+				})
+				Expect(m).To(BeNil())
+				Expect(err).To(MatchError(ContainSubstring("expected error")))
+			})
 			It("should return an error if namespace not set and not running in cluster", func() {
 				m, err := New(cfg, Options{LeaderElection: true, LeaderElectionID: "controller-runtime"})
 				Expect(m).To(BeNil())
 				Expect(err).To(HaveOccurred())
 				Expect(err.Error()).To(ContainSubstring("unable to find leader election namespace: not running in-cluster, please specify LeaderElectionNamespace"))
 			})
 
-			// We must keep this default until we are sure all controller-runtime users have upgraded from the original default
-			// ConfigMap lock to a controller-runtime version that has this new default. Many users of controller-runtime skip
-			// versions, so we should be extremely conservative here.
-			It("should default to ConfigMapsLeasesResourceLock", func() {
-				m, err := New(cfg, Options{LeaderElection: true, LeaderElectionID: "controller-runtime", LeaderElectionNamespace: "my-ns"})
+			DescribeTable("should create the correct ResourceLock", func(resourceLock string, expectMultiLock bool, expectedLocks ...resourcelock.Interface) {
+				m, err := New(cfg, Options{LeaderElection: true, LeaderElectionResourceLock: resourceLock, LeaderElectionID: "controller-runtime", LeaderElectionNamespace: "my-ns"})
 				Expect(m).ToNot(BeNil())
 				Expect(err).ToNot(HaveOccurred())
 				cm, ok := m.(*controllerManager)
 				Expect(ok).To(BeTrue())
-				multilock, isMultiLock := cm.resourceLock.(*resourcelock.MultiLock)
-				Expect(isMultiLock).To(BeTrue())
-				_, primaryIsConfigMapLock := multilock.Primary.(*resourcelock.ConfigMapLock)
-				Expect(primaryIsConfigMapLock).To(BeTrue())
-				_, secondaryIsLeaseLock := multilock.Secondary.(*resourcelock.LeaseLock)
-				Expect(secondaryIsLeaseLock).To(BeTrue())
 
-			})
+				if expectMultiLock {
+					Expect(cm.resourceLock).To(BeAssignableToTypeOf(&resourcelock.MultiLock{}))
+					multiLock := cm.resourceLock.(*resourcelock.MultiLock)
+					Expect(multiLock.Primary).To(BeAssignableToTypeOf(expectedLocks[0]))
+					Expect(multiLock.Secondary).To(BeAssignableToTypeOf(expectedLocks[1]))
+				} else {
+					Expect(cm.resourceLock).To(BeAssignableToTypeOf(expectedLocks[0]))
+				}
+			},
+				// We must keep this default until we are sure all controller-runtime users have upgraded from the original default
+				// ConfigMap lock to a controller-runtime version that has this new default. Many users of controller-runtime skip
+				// versions, so we should be extremely conservative here.
+				Entry("not specified (should default to ConfigMapsLeasesResourceLock)", "", true, &resourcelock.ConfigMapLock{}, &resourcelock.LeaseLock{}),
```

Comment:
I don't think there is value in testing more than `use default` and `use non-default` and these testcases are super annoying to read. Please just do the stupid thing, leave the existing test case how it was and add a copy of it where you explicitly configure a different Lock and verify that gets passed through

Reply:
Fair enough, done. PTAL.